src/store.cpp

```diff
@@ -0,0 +1,258 @@
+		icons.clear();
+
+		if ((attr = categoryNode.attribute("name"))) {
+			name = attr.as_string();
+		} else {
+			std::cout << "[Error - Store::loadFromXml] Missing category name." << std::endl;
+			continue;
+		}
+
+		if ((attr = categoryNode.attribute("state"))) {
+			state = static_cast<StoreOfferState_t>(pugi::cast<uint32_t>(attr.value()));
+		}
+
+		if ((attr = categoryNode.attribute("description"))) {
+			description = attr.as_string();
+		}
+
+		if ((attr = categoryNode.attribute("icons"))) {
+			icons = explodeString(attr.as_string(), ";");
+		}
+
+		categories.emplace_back(name, state, description, icons);
+		StoreCategory& category = categories.back();
+
+		for (auto offerNode : categoryNode.children()) {
+			StoreOffer& offer = category.createOffer(runningId++);
+
+			if ((attr = offerNode.attribute("name"))) {
+				offer.name = attr.as_string();
+			} else {
+				std::cout << "[Error - Store::loadFromXml] Missing offer name in category: " << category.name << std::endl;
+				category.offers.pop_back();
+				continue;
+			}
+
+			if ((attr = offerNode.attribute("type"))) {
+				offer.type = static_cast<StoreOfferType_t>(pugi::cast<uint8_t>(attr.value()));
+			}
+
+			if ((attr = offerNode.attribute("description"))) {
+				offer.description = attr.as_string();
+			}
+
+			if ((attr = offerNode.attribute("message"))) {
+				offer.message = attr.as_string();
+			}
+
+			if ((attr = offerNode.attribute("icons"))) {
+				offer.icons = explodeString(attr.as_string(), ";");
+			}
+
+			if ((attr = offerNode.attribute("price"))) {
+				offer.price = attr.as_uint();
+			}
+
+			for (auto subOfferNode : offerNode.children()) {
+				name.clear();
+				state = StoreOfferState_t::STORE_OFFERSTATE_NONE;
+				description.clear();
+				icons.clear();
+
+				if ((attr = subOfferNode.attribute("name"))) {
+					name = attr.as_string();
+				}
+
+				if ((attr = subOfferNode.attribute("state"))) {
+					state = static_cast<StoreOfferState_t>(pugi::cast<uint32_t>(attr.value()));
+				}
+
+				if ((attr = subOfferNode.attribute("description"))) {
+					description = attr.as_string();
+				}
+
+				if ((attr = subOfferNode.attribute("icons"))) {
+					icons = explodeString(attr.as_string(), ";");
+				}
+
+				offer.subOffers.push_back(SubOffer(name, state, description, icons));
+			}
+
+			if ((attr = offerNode.attribute("script"))) {
+				if (!scriptInterface) {
+					scriptInterface.reset(new LuaScriptInterface("Store Interface"));
+					scriptInterface->initState();
+				}
+
+				std::string script = attr.as_string();
+				if (scriptInterface->loadFile("data/store/scripts/" + script) == 0) {
+					offer.scriptInterface = scriptInterface.get();
+					offer.renderEvent = scriptInterface->getEvent("onRender");
+					offer.buyEvent = scriptInterface->getEvent("onBuy");
+				} else {
+					std::cout << "[Warning - Store::loadFromXml] Can not load script: " << script << std::endl;
+					std::cout << scriptInterface->getLastLuaError() << std::endl;
```

Comment:
If the reload fails you should return false, not true.
Otherwise the reload function will tell the player that the reload was successful.